### PR TITLE
Feature/fix server

### PR DESCRIPTION
**Pull request by aaron-goff**
Update documentation to reflect correct mocha syntax (#23)
Update functions to take in objects instead of just parameters
Update server to use a different process management, which will help when users start to use this library.
Update plugin to wait for headers to be generated (#22)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@willowtreeapps/rokul-runnings",
-  "version": "0.1.3",
+  "version": "0.1.4",
```
Comment (aaron-goff) [inline package.json]: I plan on publishing a new version to npm after this is merged in.

File: test/library-unit-tests.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import * as elementData from '../src/utils/elementData';
 import * as mockData from './resources/webdriver-mock-data';
 import { Buttons, Library } from '../src/modules/library';
-import assert = require('assert');
 import nock = require('nock');
+import chai = require('chai');
```
Comment (mcmillhj-wta) [inline test/library-unit-tests.ts]: should these be imports instead of require?

Comment (mcmillhj-wta) [reply test/library-unit-tests.ts]: if `chai` exports a named function called `expect` you can import it like this: 

```
import { expect } from 'chai';
```

File: test/plugin-unit-tests.ts [hunk 3]
```diff
@@ -1,13 +1,16 @@
 import { Plugin } from '../src/modules/plugin';
 import { screenshotResponse } from './resources/screenshot-response';
 import { sideloadResponse } from './resources/sideload-response';
-import nock = require('nock');
-import assert = require('assert');
+import chai = require('chai');
 import fs = require('fs');
+import nock = require('nock');
```
Comment (mcmillhj-wta) [inline test/plugin-unit-tests.ts]: should these be imports instead of require?

**Comment by aaron-goff** [comment]
> The code itself looks good but the number of times that the retry logic is repeated in various places is a little worrying. I could see them diverging accidentally. One option would be to create a `retry` utility function then delegate to that instead of repeating the logic.

Our solution is to remove the retry logic from the `http.ts` functions and add a retry method on that page.

File: test/library-unit-tests.ts (hunk 2 above)
Comment (aaron-goff) [reply test/library-unit-tests.ts]: Fixed in 59afaf3

File: test/plugin-unit-tests.ts (hunk 3 above)
Comment (aaron-goff) [reply test/plugin-unit-tests.ts]: Fixed in 59afaf3

**Pull request merged**